crates/rpc/ipc/src/server/mod.rs

```diff
@@ -68,17 +71,17 @@ impl<HttpMiddleware, RpcMiddleware> IpcServer<HttpMiddleware, RpcMiddleware> {
 
 impl<HttpMiddleware, RpcMiddleware> IpcServer<HttpMiddleware, RpcMiddleware>
 where
-    RpcMiddleware: Layer<RpcService> + Clone + Send + 'static,
-    for<'a> <RpcMiddleware as Layer<RpcService>>::Service: RpcServiceT<'a>,
-    HttpMiddleware: Layer<TowerServiceNoHttp<RpcMiddleware>> + Send + 'static,
-    <HttpMiddleware as Layer<TowerServiceNoHttp<RpcMiddleware>>>::Service: Send
-        + Service<
-            String,
-            Response = Option<String>,
-            Error = Box<dyn std::error::Error + Send + Sync + 'static>,
-        >,
-    <<HttpMiddleware as Layer<TowerServiceNoHttp<RpcMiddleware>>>::Service as Service<String>>::Future:
-    Send + Unpin,
+    RpcMiddleware: for<'a> Layer<RpcService, Service: RpcServiceT<'a>> + Clone + Send + 'static,
+    HttpMiddleware: Layer<
+            TowerServiceNoHttp<RpcMiddleware>,
+            Service: Service<
+                String,
+                Response = Option<String>,
+                Error = Box<dyn std::error::Error + Send + Sync + 'static>,
+                Future: Send + Unpin,
+            > + Send,
+        > + Send
+        + 'static,
```

Comment:
this where clause used to make rustfmt choke on the entire file

Reply:
omg